### PR TITLE
CoursesView: Activities should use their tag color when displayed

### PR DESCRIPTION
**Pull request by ltwheeler**
Currently IPA does the following for the 3 kinds of visibility/course selection
screen shot 2017-11-08 at 3 04 25 pm
screen shot 2017-11-08 at 3 04 31 pm
screen shot 2017-11-08 at 3 04 35 pm

The current change looks like this:
screen shot 2017-11-08 at 3 10 03 pm
screen shot 2017-11-08 at 3 10 08 pm
screen shot 2017-11-08 at 3 10 13 pm

IPA's current behavior is a little strange, with no visual change after selecting a sectionGroup (picture 2).

I think selecting a sectionGroup should do something, but unclear what. going from red -> gray -> teal may not be best.

### CODE REVIEW

File: app/scheduling/directives/termCalendar.js [hunk 1]
```diff
@@ -121,10 +121,29 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
 				if (scope.view.state.uiState.checkedSectionGroupIds.length > 0) {
 					scope.view.state.uiState.checkedSectionGroupIds.forEach(function (sgId) {
 						if (sgId !== scope.view.state.uiState.selectedSectionGroupId) {
+							var sectionGroup = scope.view.state.sectionGroups.list[sgId];
+							var course = scope.view.state.courses.list[sectionGroup.courseId];
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Any chance course or sectionGroup won't exist?

**Comment by cthielen** [comment]
Code looks fine, let me QA it first before merging ...

**Comment by ltwheeler** [comment]
Chris's feedback from the issue:
"It might be best to change the 'selected' activity indicator to be border-based instead.
Perhaps a 2 or 3px blue border for everything in the SG and a 4 or 5 px blue border (the grey blue IPA uses) for the highlighted activity specifically? This will need to be messed around with."

**Comment by ltwheeler** [comment]
screen shot 2017-12-01 at 5 29 31 pm

File: app/scheduling/directives/termCalendar.js [hunk 2]
```diff
@@ -104,34 +104,104 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
 				});
 			};
 
+			// supply a color and amount to shift it by
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Add a comment as to the units we're shifting by. Are these percents? Or are they out of 255?

File: app/scheduling/directives/termCalendar.js [hunk 3]
```diff
@@ -104,34 +104,104 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
 				var calendarActivities = [];
 
 				// Add Selected sectionGroup activities
 				if (scope.view.state.uiState.selectedSectionGroupId) {
-					var unstyledEvents = sectionGroupToActivityEvents(scope.view.state.sectionGroups.list[scope.view.state.uiState.selectedSectionGroupId]);
+					var sectionGroup = scope.view.state.sectionGroups.list[scope.view.state.uiState.selectedSectionGroupId];
+					var unstyledEvents = sectionGroupToActivityEvents(sectionGroup);
+					var tagColor = calculateTagColor(sectionGroup);
+
+					var textColor = tagColor ? "#FFFFFF" : activeEventTextColor;
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Define a variable for whatever "#FFFFFF" stands for, e.g. inactiveEventTextColor ?

File: app/scheduling/directives/termCalendar.js [hunk 4]
```diff
@@ -104,34 +104,104 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
+								);
+							}
 						}
 					});
 				}
 
 				return calendarActivities;
 			};
 
+			var calculateTagColor = function (sectionGroup) {
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Add a comment indicating what this calculates based on, e.g. takes the color of the first tag, if any, else returns null?

File: app/scheduling/directives/termCalendar.js [hunk 5]
```diff
@@ -104,34 +104,104 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
+				var tagColor = null;
+				var course = scope.view.state.courses.list[sectionGroup.courseId];
+
+				if (course.tagIds.length > 0) {
+					var tag = scope.view.state.tags.list[course.tagIds[0]];
+					if (tag.color) {
+						tagColor = tag.color;
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Do we want to return early once tagColor is set? Or are we taking the last tag color?

File: app/scheduling/directives/termCalendar.js [hunk 6]
```diff
@@ -253,11 +323,24 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
 				return calendarActivities;
 			};
 
-			var styleCalendarEvents = function (calendarActivities, backgroundColor, borderColor, textColor) {
+			var styleCalendarEvents = function (calendarActivities, backgroundColor, borderColor, textColor, tagColor) {
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Add a comment indicating what this function is doing, e.g. sets text color, border Color, and background color based on UI selection state, etc.

File: app/scheduling/directives/termCalendar.js [hunk 7]
```diff
@@ -253,11 +323,24 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
 				calendarActivities.forEach(function (event) {
-					event.color = (scope.view.state.uiState.selectedActivityId === event.activityId) ? highlightedEventBackgroundColor : backgroundColor;
-					event.borderColor = (scope.view.state.uiState.selectedActivityId === event.activityId) ? highlightedEventBorderColor : borderColor;
-					event.textColor = (scope.view.state.uiState.selectedActivityId === event.activityId) ? highlightedEventTextColor : textColor;
+					console.log(scope.view.state.uiState.selectedActivityId);
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Remove this console.log().

File: app/scheduling/directives/termCalendar.js [hunk 8]
```diff
@@ -253,11 +323,24 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
+					if (scope.view.state.uiState.selectedActivityId === event.activityId) {
+						if (tagColor) {
+							event.color = lightenOrDarkenColor(tagColor, -80);
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Define a constant for "-80" which is named to explain what it is, e.g. SELECTED_ACTIVITY_COLOR_SHIFT or something better.

File: app/scheduling/directives/termCalendar.js [hunk 9]
```diff
@@ -104,27 +107,83 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
...
 				if (scope.view.state.uiState.checkedSectionGroupIds.length > 0) {
 					scope.view.state.uiState.checkedSectionGroupIds.forEach(function (sgId) {
 						if (sgId !== scope.view.state.uiState.selectedSectionGroupId) {
+							var sectionGroup = scope.view.state.sectionGroups.list[sgId];
+							var course = scope.view.state.courses.list[sectionGroup.courseId];
 							var unstyledEvents = sectionGroupToActivityEvents(scope.view.state.sectionGroups.list[sgId]);
-							calendarActivities = calendarActivities.concat(
-								styleCalendarEvents(unstyledEvents)
-							);
+
```
Comment (cthielen) [inline app/scheduling/directives/termCalendar.js]: Extra blank line here

**Pull request merged**